test/controllers/products_controller_test.rb

```diff
@@ -2,28 +2,145 @@
 
 describe ProductsController do
   let(:existing_product) { products(:turtleneck) }
+  let(:new_merchant) { users(:merchant) }
 
   it "should get index" do
     get products_path
- 
+
     must_respond_with :success
   end
 
-  # it "should get show" do
-  #   get products_show_url
-  #   value(response).must_be :success?
-  # end
+  describe "show" do
+    it "should be OK to show an existing, valid product" do
+      valid_product_id = existing_product.id
 
-  # it "should get new" do
-  #   get products_new_url
-  #   value(response).must_be :success?
-  # end
+      get product_path(valid_product_id)
 
-  # it "should get create" do
-  #   get products_create_url
-  #   value(response).must_be :success?
+      must_respond_with :success
+    end
+
+    it "should give a flash notice instead of showing a non-existant, invalid product" do
+      product = existing_product
+      invalid_product_id = product.id
+      product.destroy
+
+      get product_path(invalid_product_id)
+
+      must_respond_with :not_found
+      expect(flash[:error]).must_equal "Could not find product with id: #{invalid_product_id}"
+    end
+  end
+
+  # describe "new" do
+  #   it "succeeds" do
+  #     get new_product_path
+
+  #     must_respond_with :success
+  #   end
   # end
 
+  describe "create" do
+    describe "logged in users" do
+      before do
+        perform_login(users(:dee))
+        merchant = new_merchant
+      end
+
+      it "will save a new product and redirect if given valid inputs" do
+        input_photo_url = "https://drive.google.com/uc?id=1LCGn0419g0STAeyDo-miWCD6o5ZOEkXM"
+        input_description = "black, lightweight, breathable wool turtleneck, perfect for those moments when you're sweating your scam."
+        input_name = "Iconic Tech Turtleneck"
+        input_price = 9900
+        input_quantity = 20
+        input_retired = false
```

Comment:
I think we could get rid of this line cause we already set the default to false but not 100% sure.

Reply:
totally - deleted!